### PR TITLE
Std out sink to use event model

### LINKED ISSUE #585: StdOut Sink to use Event Model
The stdout sink is a good way to get started and test the grok prepper. The recent 
Event model change #538 broke the connection between the stdout sink and grok, as it still uses a `Record<String>` type while grok was updated to a `Record<Event>`

### PR DESCRIPTION
**Pull request by graytaylor0**
### Description
Updated stdout sink to use the Event model
 
### Issues Resolved
Closes #585 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java [hunk 1]
```diff
@@ -12,21 +13,47 @@
...
 
 public class StdOutSinkTests {
     private static String PLUGIN_NAME = "stdout";
 
-    private final String TEST_DATA_1 = "data_prepper";
-    private final String TEST_DATA_2 = "stdout_sink";
-    private final Record<String> TEST_RECORD_1 = new Record<>(TEST_DATA_1);
-    private final Record<String> TEST_RECORD_2 = new Record<>(TEST_DATA_2);
-    private final List<Record<String>> TEST_RECORDS = Arrays.asList(TEST_RECORD_1, TEST_RECORD_2);
+    private final Map<String, Object> TEST_DATA_1 = new HashMap<>();
```
Comment (dlvenable) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:31]: I recommend that you make this a non-final and initialize it in the `setup`. As it is right now, authors will need to clear the map in-between tests.

The same goes for `TEST_DATA_2` and `TEST_RECORDS`.

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java [hunk 2]
```diff
@@ -12,21 +13,47 @@
 package com.amazon.dataprepper.plugins.sink;
 
 import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
 import com.amazon.dataprepper.model.record.Record;
 import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
```
Comment (cmanning09) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:20]: You are using two different version of Junit now. I believe this is what is impacting your test coverage.

**Review by cmanning09 — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/common/build.gradle [hunk 3]
```diff
@@ -32,7 +32,7 @@ jacocoTestCoverageVerification {
     violationRules {
         rule { //in addition to core projects rule
             limit {
-                minimum = 0.90
+                minimum = 0.89
```
[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/common/build.gradle:35]: You would think so, but the coverage is now .89 after this PR

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java (hunk 2 above)
Comment (graytaylor0) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:20]: Interesting

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java [hunk 4]
```diff
@@ -12,26 +13,59 @@
...
 
 public class StdOutSinkTests {
     private static String PLUGIN_NAME = "stdout";
 
-    private final String TEST_DATA_1 = "data_prepper";
-    private final String TEST_DATA_2 = "stdout_sink";
-    private final Record<String> TEST_RECORD_1 = new Record<>(TEST_DATA_1);
-    private final Record<String> TEST_RECORD_2 = new Record<>(TEST_DATA_2);
-    private final List<Record<String>> TEST_RECORDS = Arrays.asList(TEST_RECORD_1, TEST_RECORD_2);
+    List<Record<Object>> TEST_RECORDS;
```
Comment (dlvenable) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java:31]: This field and the variables in `setup` should be renamed to standard Java field and variable conventions.

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks for making it support both! I have one comment on variable names and then I'll be ready to approve. It looks like the 90% coverage is working again, correct?

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java (hunk 2 above)
Comment (dlvenable) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java]: Good catch! It's a subtle difference.

File: data-prepper-plugins/common/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmanning09) [reply data-prepper-plugins/common/build.gradle]: I ran your changes locally addressing the Junit issue and looking at the code coverage results from the build. It is possible to achieve .90. Let's keep it at .90.

[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/common/build.gradle]: I had changed tried this yesterday and it did fix it. Had bumped it back to 0.90 for the next revision. Interesting little thing there with the 2 different versions of Junit. Thanks for catching it!

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java (hunk 2 above)
Comment (graytaylor0) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/StdOutSinkTests.java]: > Thanks for making it support both! I have one comment on variable names and then I'll be ready to approve. It looks like the 90% coverage is working again, correct?

Yep .90 is working again

**Review by dlvenable — APPROVED** [review]

**Review by cmanning09 — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**